### PR TITLE
Removes Pump Up, Therefore Removing Ghetto Counters to New Batons

### PR DESCRIPTION
**Pull request by wesoda25**
## About The Pull Request

Removes pump up, restoring chemical counters to batons to what they formerly were, adrenals and certain drugs such as meth. 

## Why It's Good For The Game

Batons are meant to be a fail-safe for a security. A reliable weapon whose use is consistent and effective. But with this new change, they are anything but. By walking around maint, one can find pump up injectors to shrug off the effects of a baton. Or, through minimal investment, manufacture some of the counter for themselves. I believe this is bad gameplay, and encourages powergaming. We already have a system of sort in place for stun resist, and while it might need some tweaking with these new changes, I think we will benefit overall in the time being by removing pump up (not to say we'll re add it after though, it should stay gone). Batons are ideal in the state of leaving a small window of opportunity to react. 

As an aside to Akrilla: I hope theres no hard feelings over this PR. I fucking love what you've done with batons, but I really don't think the chem stuff should stay. I know its your design vision but I think its for the best they get ganked. 

## Changelog
:cl:
del: Removes pump up. This means no more ghetto counters to stun batons can be found, or created. Invest in adrenals if you wanna shrug off a baton!
/:cl:

### CODE REVIEW

File: code/game/objects/items/stunbaton.dm [hunk 1]
```diff
@@ -193,13 +193,8 @@
 
 /// After the initial stun period, we check to see if the target needs to have the stun applied.
 /obj/item/melee/baton/proc/apply_stun_effect_end(mob/living/target)
```
Comment (wesoda25) [inline code/game/objects/items/stunbaton.dm]: Is this necessary to leave in, or can I remove this bit entirely?

**Comment by optimumtact** [comment]
what if im already happy with the new meta for batons and your old one no longer applies?

**Comment by wesoda25** [comment]
When you say new meta what do you mean though? You enjoy the baton in its current state of being ineffective against not only numerous types of antags but against assistants as well? 

I simply think the results should be consistent and not inconsistent, so long as the victim invested their TC/amount of time/ling points into some sort of counter to stuns. Pump up also displaces the “old meta” of those things being what they are, as it renders them pointless since anyone with access to a vending machine and a medical fridge can easily bypass what they’d pay for the respective stun counters. 

Its not so much a meta/what this item is meant to do argument, its an argument that what I’m proposing is better than what is in place atm. If your “new meta” or design vision for batons conflicts with that then I guess I’m fucked, aren’t I?

**Comment by optimumtact** [comment]
ineffective huh

**Comment by wesoda25** [comment]
Yep. You can baton someone and they won’t even drop. That basically means stun batons are a slowdown stick, which is pointless since as sec you already have a ranged slowdown gun (referring to disabler not temp gun). Its quite literally rewarding players for taking illogical precautions, just in case they happen to be batonned at some point. And in the traitor sense, where it is a logical precaution, it allows them to save some tc or whatever antag currency they have.  

Also whats your definition of the new baton meta?

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 2]
```diff
@@ -440,46 +440,3 @@
...
 	..()
 	. = 1
-
-/datum/reagent/drug/pumpup
-	name = "Pump-Up"
-	description = "Take on the world! A fast acting, hard hitting drug that pushes the limit on what you can handle."
-	reagent_state = LIQUID
-	color = "#e38e44"
-	metabolization_rate = 2 * REAGENTS_METABOLISM
-	overdose_threshold = 30
```
Comment (JJRcop) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: you might be better off messing with this number

Comment (wesoda25) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: OD counters are good for ingrained chemicals that won’t be removed. Chemicals that have multiple uses, beneficial ones, but are seen frequently abused for combat use. Pump up exists to literally just be used a combat chem with no uses whatsoever ourside of that. Therefore I see no reason to not remove it. Plus even with OD stuff people can just manage chem amount and abuse it that way.

**Comment by necromanceranne** [comment]
There is one core problem with this wesoda. The old chems don't do anything constructive to protect you against new batons because the new batons do heavy stamina damage, which just isn't adequately counteracted by stimulants whatsoever. A stimpack doesn't even protect you from a followup smack unless you can somehow sprint off before the stun hits. By then the stamina damage has likely put you into stamina crit.

Pumpup is a bridge chem until someone updates the old chems. Until then, they're actually not in a good spot at all and aren't reliably protecting against stun batons, so they might as well not exist and you just simply lose to the first hit. Adrenals and stimulants are a total waste of TC.

**Comment by wesoda25** [comment]
They still provide some use, and the baton cooldown is helpful. However I agree they need to be updated.

**Comment by kriskog** [comment]
Pump up is a very specific chem against ONLY the hard "old" stun from specifically only the stunbaton. It does not protect against the stamina damage, so even if you get up from the first hit you will be slowed. If you want to consistently keep people down, baton them twice and put them in stam crit.

Having a counter to not only security, but also traitors that stole a stun stick is a good thing.

**Comment by wesoda25** [comment]
I disagree.

**Comment by kriskog** [comment]
> 
> Batons are meant to be a fail-safe for a security. A reliable weapon whose use is consistent and effective. But with this new change, they are anything but.

Improvise. Adapt.

**Comment by wesoda25** [comment]
Adapt to what? A broken balance of security and antagonists that somehow evens out due to oversights and forgotten code? And what does adapt mean? I now have to powergame Pump Up if I want a chance as a sec officer, or antag? Fuck that.

**Comment by necromanceranne** [comment]
I mean, removing pump up swings this balance hard in the other direction. I genuinely think the new batons are stronger than they've been in the past but I get that's contentious.

**Comment by wesoda25** [comment]
I think batons in and of themselves are balanced just bc while there is a window of opportunity to react, they can stamina spam

**Comment by Dawson1917** [comment]
I still maintain that, if there's going to be an accessible resist against stunbatons, it should be epipens. Putting it behind mixing coffee and epinephrine to make a new reagent means the counter is exclusively available to greytiders and powergamers who _expect_ to be batonned. Everyone else is still as defenseless against stunbatons as before. If the chem isn't accessible to everyone then it absolutely should be removed.

>Invest in adrenals if you wanna shrug off a baton! 

Adrenals meta is cancer and infests nearly every non-simplemob antag. Nobody likes it. Nobody needs more reasons to use them.

**Comment by necromanceranne** [comment]
React how? The delayed stun only means that if you hit first you lose if your opponent ALSO has a baton and hit you retaliation, but if you didn't have a baton yourself you lose anyway because adrenals don't do enough and stimulant drugs do fucking nothing at all short of Bathsalts.

Its' extremely arse backwards.

**Comment by wesoda25** [comment]
> React how? The delayed stun only means that if you hit first you lose if your opponent ALSO has a baton and hit you retaliation, but if you didn't have a baton yourself you lose anyway because adrenals don't do enough and stimulant drugs do fucking nothing at all short of Bathsalts.
> 
> Its' extremely arse backwards.

well, theres a lot you could do. the stun delay gives sec more time to shout for help on radio, pull out a baton of their own, etc. Its especially useful in baton v baton showdowns, its less whoever clicks first win now. if you are a triator you could whip out a gun and hurt the officer, pop stimulants, or shout for help on your own comms. Still I don't really think its a topic for discussion on this PR

**Comment by remuluson2** [comment]
honestly i love how everyone forgets about aranesp(or what ever is called) that heals 18 stamina damage per tick!

**Comment by kriskog** [comment]
> 
> Still I don't really think its a topic for discussion on this PR

You dont think a stun baton balance PR  is the right place to discuss stun baton balance?

**Comment by Akrilla** [comment]
I'm in the middle of tweaking existing combat chems to be more effective against the now stamina stun baton. It's my opinion that having a way of shrugging off these effects in a way that's *fairly* easy to acquire (thus providing a counter) provides more interesting combat than what exists. I've made my arguments in the last PR, and I'm more than a little tired of repeating them.

I think the baton needs some very slight tweaks but removing the entire chem defeats the overall vision for it. We may as well revert it back. I don't think the baton is in a bad place currently, and do believe, as do others, that it's in a better place than before. If anything I want to buff this chemical very slightly to provide more counter opportunity. 

Not going to reply more than this 🤷‍♂

**Comment by Garen7** [comment]
>I don't think the baton is in a bad place currently, and do believe, as do others, that it's in a better place than before.

I think you'd be hard pressed to prove that the majority agree with you on that.

**Comment by kriskog** [comment]
Luckily, only the right people have to agree :ok_hand:

**Comment by Akrilla** [comment]
> Yep. You can baton someone and they won’t even drop.

I will say that it's a bit worrying that you're making this PR when you say stuff like this - which just isn't true.



> > I don't think the baton is in a bad place currently, and do believe, as do others, that it's in a better place than before.
> 
> I think you'd be hard pressed to prove that the majority agree with you on that.

It's only been in the game in its current state for about a week, and only a few days for ones like Terry.

File: code/game/objects/items/stunbaton.dm (hunk 1 above)
Comment (Akrilla) [reply code/game/objects/items/stunbaton.dm]: yes, needed for delay

**Comment by carlarctg** [comment]
oh damn, let's keep encouraging traitors, nukeops and lings to always take adrenals and adrenaline sacs every single round, what a wonderful idea

stun batoning someone pumped up still drops people for a second and slows them, and hitting a limb with it even once disables it, which extra slows if you're aiming for a leg. if you _really_ have this much of a problem against greytiders you can literally just hit them in the leg, walk back 2 tiles, and shoot them with your disabler as pumpup does nothing against stamina damage anyways. it's not hard

**Comment by necromanceranne** [comment]
The bad men spent gamer points to stop my baton and this makes me sad.

It's almost like historically stuns and by extension batons were so stupidly powerful that you run an incredible risk not taking them.

They don't do much into the new batons however. They are weaker by a large margain. It's really only pumpup that properly combats the new batons.

**Comment by ExcessiveUseOfCobblestone** [comment]
I'd like the stun baton stun removed and replaced with a knockdown (the stamina damage stun should be the only stunning aspect of it imo).

**Comment by ExcessiveUseOfCobblestone** [comment]
I don't have any interest in this PR in terms of it's focus to pump-up. I think pump up is fine as-is.

**Comment by Akrilla** [comment]
> I'd like the stun baton stun removed and replaced with a knockdown (the stamina damage stun should be the only stunning aspect of it imo).

Yeah I think I should of done that in the first place

**Comment by wesoda25** [comment]
OK I won’t bother quoting them all but I see most comments are based off how bad stun combat is, and how this addresses it. But I don’t think it addresses it in a good way that should be encouraged or kept in the game. It encourages powergame and thats just actually awful. 

If, it were to be something everyone starts with, like epinephrine, I’d be less likely to call it powergame. But to go above and beyond that, ideally, if we really want it based off chem based counters, I say it should be a separate chem, impossible to create, that everyone starts with. A separate medipen, if you will. There’d be no powergame as everyone has an equal amount, and can’t get anymore (aside from looting others). 

But at the end of the day I just don’t like the idea of chem based counters alone. New batons alone are ideal compared to what we formerly had. I like cobbys idea of an instant knockdown + 80 adren damage or something, as it would have no place to have a chem counter system since the baton hit itself leaves numerous opportunities to counter.

**Comment by kriskog** [comment]
I'm not sure where you got this idea that preparation should not be beneficial. Heck, you barely even have to prepare. Ten steps into any maint tunnel and youll find an injector.

**Comment by ATH1909** [comment]
just hit them twice with your baton smh

**Comment by wesoda25** [comment]
> > Yep. You can baton someone and they won’t even drop.
> 
> I will say that it's a bit worrying that you're making this PR when you say stuff like this - which just isn't true.

I will say this did happen to me. Really not sure why, but I batoned someone and they didn't drop. I asked and they said they were on pump up, might have been combining with other stun resists and shit tho? Looking at code you are indeed right, I tried replicating and couldn't, so sorry for that claim. 

That being said they do go down for a very short amount of time which is really in the same vein.

**Comment by Ratvar** [comment]
What if everyone always was on effect of pump up? That would make stunbatons into much better weapons overall

**Comment by carlarctg** [comment]
wait are you that one assistant shitter in security i robusted as a horse mask lizard

literally an i ded pr

**Comment by Garen7** [comment]
@carlarctg if you don’t have anything useful or relevant to working on the Pr then don’t post, a thumbs down is sufficient to say you don’t like a Pr

**Comment by wesoda25** [comment]
At this point this PR is outdated. I detest pump up as a chem, but akrillas PR is more up to date. I was actually nearly done with a PR similar to theirs (minus pump up) but they PRed it first. I think I will either test the waters with their PR to just see it in effect, or I might just open a new PR.

**Pull request closed without merging**